crates/editor/src/element.rs

```diff
@@ -1635,6 +1635,101 @@ impl EditorElement {
     }
 
+    #[allow(clippy::too_many_arguments)]
+    fn layout_inline_diagnostics(
+        &self,
+        line_layouts: &[LineWithInvisibles],
+        crease_trailers: &[Option<CreaseTrailerLayout>],
+        content_origin: gpui::Point<Pixels>,
+        scroll_pixel_position: gpui::Point<Pixels>,
+        start_row: DisplayRow,
+        end_row: DisplayRow,
+        line_height: Pixels,
+        em_width: Pixels,
+        style: &EditorStyle,
+        window: &mut Window,
+        cx: &mut App,
+    ) -> HashMap<DisplayRow, AnyElement> {
+        let diagnostics = self.editor.update(cx, |editor, _| {
+            if !editor.show_inline_diagnostics() {
+                return Default::default();
+            }
+            editor
+                .inline_diagnostics
+                .range(start_row..=end_row)
+                .map(|(point, diag)| (*point, diag.clone()))
+                .collect::<Vec<_>>()
+        });
+
+        if diagnostics.is_empty() {
+            return Default::default();
+        }
+
+        let sev_to_color = |sev: &DiagnosticSeverity| match sev {
+            &DiagnosticSeverity::ERROR => Color::Error,
+            &DiagnosticSeverity::WARNING => Color::Warning,
+            &DiagnosticSeverity::INFORMATION => Color::Info,
+            &DiagnosticSeverity::HINT => Color::Hint,
+            _ => Color::Error,
+        };
+
+        let padding = ProjectSettings::get_global(cx)
+            .diagnostics
+            .inline()
+            .padding() as f32
+            * em_width;
+
+        let min_x = ProjectSettings::get_global(cx)
+            .diagnostics
+            .inline()
+            .min_column() as f32
+            * em_width;
+
+        let mut elements = HashMap::default();
+        for (row, diagnostic) in diagnostics {
+            if !(start_row..end_row).contains(&row) {
+                continue;
+            }
+
+            let pos_y = content_origin.y
+                + line_height * (row.0 as f32 - scroll_pixel_position.y / line_height);
+
+            let window_ix = (row - start_row).0 as usize;
+            let pos_x = {
+                let crease_trailer_layout = crease_trailers[window_ix].as_ref();
+                let line_layout = &line_layouts[window_ix];
+
+                let line_end = if let Some(crease_trailer) = crease_trailer_layout {
+                    crease_trailer.bounds.right()
+                } else {
+                    content_origin.x - scroll_pixel_position.x + line_layout.width
+                };
+
+                let padded_line = line_end + padding;
+                let min_start = content_origin.x - scroll_pixel_position.x + min_x;
+
+                cmp::max(padded_line, min_start)
+            };
+
+            let mut element = h_flex()
+                .id(SharedString::from(format!("diagnostic-{}", row.0)))
+                .h(line_height)
+                .w_full()
+                .bg(sev_to_color(&diagnostic.severity).color(cx).opacity(0.07))
```

Comment:
```suggestion
                .bg(sev_to_color(&diagnostic.severity).color(cx).opacity(0.07))
                .px_1()
```
I really like how this feels now! Can we add a small padding here? I think that makes it a little bit nicer.

| Before | After |
|-----------|-----------|
| <img width="435" alt="before" src="https://github.com/user-attachments/assets/9c177cf1-d60b-470b-84f4-7c675138eeb2" /> | <img width="435" alt="after" src="https://github.com/user-attachments/assets/19bb45f2-f2d1-467b-a482-915bce3504bc" /> |